wallet/createtx.go

```diff
@@ -286,11 +287,30 @@ func (w *Wallet) findEligibleOutputs(dbtx walletdb.ReadTx,
 		if addrAcct != account {
 			continue
 		}
+
+		// Skip inputs that do not raise the total transaction output
```

Comment:
I don't think we should have this type of logic at this layer, instead it should be up to the caller to decide if these inputs should be skipped or not. As implemented, this may lead to a change in behavior resulting in outputs that would've been spent as normal, now requiring manual code to fetch and sign for in order to sweep.

Reply:
Moved filtering for positively yielding inputs to the caller `Wallet.txToOutputs`. But not completely sure if this is what you meant though. Do you mean a change in behavior for future other users of `findEligibleOutputs`?

Reply:
I mean that there're classes of transactions the _existing_ code would produce, that this no longer will by default (since it enforces only positive yield inputs being added). As a result, people wouldn't be able to use the normal "sweep all" behavior to clear _all_ UTXOs independent of the net cost/yield (they just want to pay down their UTXO debt and aren't sensitive to the cost. 

By "this layer" I meant that imo this should be up to the caller if this should be done or not. Every once in a while when fees dip low, I have people reach out to make to figure out how to clear out all their fragmented UTXOs. With this change (to only every include inputs with positive yield), they'd have to write a custom script using the sub-servers to sweep things.

Reply:
Ah, got it. Updated branch to keep the old behavior unchanged.

It is unclear to me though why users would want to spend money to clear out fragmented utxos.

Reply:
The main reason at the moment is that the current 'largest' coin selection strategy produces an ever-increasing UTXO count of dust like transactions... when you end up with a massive number of UTXOs it's hard to get a feel for how much of it is unspendable, how much to consolidate when blockchain fees are low, and how bad of a problem this really is. So it's good to be able to reset the UTXO count back to 1, even if some UTXOs are wasted in the process.

I guess this use-case will go away once we can use the 'random' strategy though... or hopefully an even better strategy in the future? Ideally I'd lilke to target an overall UTXO count range, and the strategy would do the most intelligent thing possible to keep it near that.